### ISSUE TITLE
Getting wrong error "Strings not allowed in JSX files" in prop

### ISSUE THREAD

**Issue by iiison** [yannickcr/eslint-plugin-react#2142]
I'm getting error in this:
```js
<div className={`col-12 grid input-cont ${shouldUseDefaultClasses && defaultContClass} ${contClass}`}>

```

The error is:
```sh
error  Strings not allowed in JSX files: “`col-12 grid input-cont ${shouldUseDefaultClasses && defaultContClass} ${contClass}`”  react/jsx-no-literals
```

There is no string in the className, not sure what am I missing here.

**Comment by ljharb** [comment]
This should definitely be fine since it's in an attribute.

**Comment by iiison** [comment]
not sure what's wrong. Here is my package versions:

| package              | Version |
|----------------------|---------|
| babel-eslint         | 10.0.1  |
| eslint               | 5.12.1  |
| eslint-config-airbnb | 17.1.0  |
| eslint-loader        | 2.1.1   |
| eslint-plugin-react  | 7.12.4  |

**Comment by ljharb** [comment]
What's your eslint config?

**Comment by iiison** [comment]
Here is [link](https://gist.github.com/iiison/e3a88fa1e1f8a62a5a85d3523360aee1)

**Comment by ljharb** [comment]
k, looks like a legitimate bug

**Comment by iiison** [comment]
hmm, maybe today is my bad day, another strange thing is happening: 

if I use:

```jsx
    items = <li className='suggestions col-12 lighter-color'>{'No matching suggestion.'}</li>
```

I get error:
```sh
Curly braces are unnecessary here.
```

and I remove curly braces like this:

```jsx
<li className='suggestions col-12 lighter-color'>'No matching suggestion.'</li>
```

I get error:
```sh
Strings not allowed in JSX files: "No matching suggestion"
```

**Comment by ljharb** [comment]
Well that's true - the "no literal" rule explicitly forbids string literals. (it's not in curly braces, so it's a string literal)

**Comment by iiison** [comment]
how do I fix it then.. can't keep the curly braces can't remove them either..

**Comment by ljharb** [comment]
We'll need a PR to this plugin to fix it.

**Comment by iiison** [comment]
can I create a PR..

**Comment by ljharb** [comment]
Absolutely!

**Comment by iiison** [comment]
@ljharb as you can see in my [config](https://gist.github.com/iiison/e3a88fa1e1f8a62a5a85d3523360aee1) file, I am using airbnb style guid too, and there the rule for `jsx-no-literals` is this: 
```js
'react/jsx-no-literals': ['off', { noStrings: true }],
``` 

and in my config file I've this rule: 
```js
'react/jsx-no-literals': 2
```
Everything works fine if I make `noStrings` to false in config. does that mean eslint doesn't overwrite the whole rule and I've to define write options too in order to truly overwrite the rule from plugin?

Not sure if it's a bug or kept it this way for a reason.

**Comment by iiison** [comment]
one more thing, should we have a little more details about the API and how the control flows in contributing file. LMK if I could be a help..

**Comment by ljharb** [comment]
Sure, a separate PR to improve CONTRIBUTING.md is always great.

As for `noStrings`; https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-literals.md says "Enforces no string literals used as children, wrapped or unwrapped." - that means it should only apply to children, not props, which seems like the bug.

**Comment by iiison** [comment]
@ljharb I've created a [pull request](https://github.com/yannickcr/eslint-plugin-react/pull/2146) for the issue, but when I see in [test file](https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/jsx-no-literals.js#L303-L344), it doesn't allow strings in props. I've updated the test in PR tho. Just wanted to make sure if we are on the same page.

**Comment by pawelnvk** [comment]
Is this something I could start working with? Or is it WIP?

**Comment by ljharb** [comment]
@pawelnvk #2146 already exists

**Comment by pawelnvk** [comment]
My mistake 😄 Thanks.

**Comment by jzabala** [comment]
> @pawelnvk #2146 already exists

Should this issue be closed then?

**Comment by ljharb** [comment]
Yep, now that it's merged. Thanks!
